includes/admin/class-amp-admin-pointer.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/**
+ * Admin pointer class.
+ *
+ * @package AMP
+ * @since 1.0
+ */
+
+/**
+ * AMP_Admin_Pointer class.
+ *
+ * Outputs an admin pointer to show the new features of v1.0.
+ * Based on https://code.tutsplus.com/articles/integrating-with-wordpress-ui-admin-pointers--wp-26853
+ *
+ * @since 1.0
+ */
+class AMP_Admin_Pointer {
+
+	/**
+	 * The ID of the template mode admin pointer.
+	 *
+	 * @var string
+	 */
+	const TEMPLATE_POINTER_ID = 'amp_template_mode_pointer_10';
+
+	/**
+	 * The slug of the script.
+	 *
+	 * @var string
+	 */
+	const SCRIPT_SLUG = 'amp-admin-pointer';
+
+	/**
+	 * Initializes the class.
+	 *
+	 * @since 1.0
+	 */
+	public function init() {
+		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_pointer' ) );
+	}
+
+	/**
+	 * Enqueues the pointer assets.
+	 *
+	 * If the pointer has not been dismissed, enqueues the style and script.
+	 * And outputs the pointer data for the script.
+	 *
+	 * @since 1.0
+	 */
+	public function enqueue_pointer() {
+		$dismissed = explode( ',', strval( get_user_meta( get_current_user_id(), 'dismissed_wp_pointers', true ) ) );
+
+		// Exit if the pointer has been dismissed.
+		if ( in_array( self::TEMPLATE_POINTER_ID, $dismissed, true ) ) {
```

Comment:
Here, this can now be simplified to this:

```
		if ( $this->has_pointer_been_dismissed() ) {
			return;
		}
```

These changes make it to read and quickly know what's going on for the guard clause.

Reply:
Thanks, this is applied in commit [e60f0](https://github.com/Automattic/amp-wp/pull/1271/commits/e60f0f67fced870d11e9e87690198cc32ad39db8#diff-7ec3f55e1310503aa82ce61984d3f482R51).